src/west/commands/project.py

```diff
@@ -212,38 +212,142 @@ def __init__(self):
             _wrap('''
             Updates all projects according to the manifest file, `west.yml`,
             in the manifest repository.
-            All projects will be checked out in detached HEAD and the local
-            manifest-rev branch will be reset hard to same HEAD.
 
-            Any local branches will not be altered by this command.
+            By default:
 
-            This command will not touch the manifest repository.
+            1. West itself is updated to the newest revision in the manifest
+            file.
+
+            2. The local per-project manifest-rev branches tracked by
+            west will be hard reset to the updated commits fetched
+            from each project's remote.
```

Comment:
Should it be mentioned the the branch `manifest-rev` is a west maintained branch, and that no work should be done on this branch ?

Reply:
That's what I am trying to say with "per-project manifest-rev branches tracked by west".

Would it be more clear if this said "per-project manifest-rev branches, which are managed exclusively by west"?

Reply:
I think that highlights better that it's not to be checked out by user.